### ISSUE TITLE
[CRITICAL] unable to access item 'randomStr'

### ISSUE THREAD

**Issue by stayliv3** [sqlmapproject/sqlmap#1879]
git pull the newest version. and got the error [CRITICAL] unable to access item 'randomStr'

```
➜  sqlmap git:(master) ✗ python sqlmap.py -r request/tom1.txt --random-agent -o --prefix="-1'" --suffix="%20AND%20000847%3d000847%20or%20'axslDyzU'%3d'" --technique=B --risk 3
         _
 ___ ___| |_____ ___ ___  {1.0.5.36#dev}
|_ -| . | |     | .'| . |
|___|_  |_|_|_|_|__,|  _|
      |_|           |_|   http://sqlmap.org

[!] legal disclaimer: Usage of sqlmap for attacking targets without prior mutual consent is illegal. It is the end user's responsibility to obey all applicable local, state and federal laws. Developers assume no liability and are not responsible for any misuse or damage caused by this program

[*] starting at 11:11:33

[11:11:33] [INFO] parsing HTTP request from 'request/tom1.txt'
[11:11:33] [INFO] fetched random HTTP User-Agent header from file '/Users/litsand/security/web/sqlmap/txt/user-agents.txt': 'Mozilla/5.0 (Windows; U; Windows NT 6.1; et; rv:1.9.1.9) Gecko/20100315 Firefox/3.5.9'
custom injection marking character ('*') found in option '--data'. Do you want to process it? [Y/n/q] y
[11:11:35] [INFO] testing connection to the target URL
[11:11:35] [INFO] testing if the target URL is stable
[11:11:36] [INFO] target URL is stable
[11:11:36] [INFO] testing if (custom) POST parameter '#1*' is dynamic
[11:11:36] [WARNING] (custom) POST parameter '#1*' does not appear dynamic
[11:11:36] [WARNING] heuristic (basic) test shows that (custom) POST parameter '#1*' might not be injectable
[11:11:37] [INFO] testing for SQL injection on (custom) POST parameter '#1*'
[11:11:37] [INFO] testing 'AND boolean-based blind - WHERE or HAVING clause'
[11:11:37] [INFO] testing 'OR boolean-based blind - WHERE or HAVING clause'
[11:11:38] [CRITICAL] unable to access item 'randomStr'
```

**Comment by martindube** [comment]
Having this issue too with:

./sqlmap.py -r something.txt -p login --string "something" --prefix "'" --suffix "##" --dbms=mysql --tamper=between,space2comment --technique=B

[14:34:09] [INFO] parsing HTTP request from 'something.txt'
[14:34:09] [INFO] loading tamper script 'between'
[14:34:09] [INFO] loading tamper script 'space2comment'
[14:34:09] [INFO] testing connection to the target URL
[14:34:11] [INFO] testing if the provided string is within the target URL page content
[14:34:13] [WARNING] you provided 'something' as the string to match, but such a string is not within the target URL raw response, sqlmap will carry on anyway
[14:34:13] [WARNING] heuristic (basic) test shows that POST parameter 'login' might not be injectable
[14:34:13] [INFO] testing for SQL injection on POST parameter 'login'
[14:34:13] [INFO] testing 'AND boolean-based blind - WHERE or HAVING clause'
[14:34:44] [INFO] testing 'OR boolean-based blind - WHERE or HAVING clause'
[14:34:49] [CRITICAL] unable to access item 'randomStr'

**Comment by stamparm** [comment]
@stayliv3 @martindube really can't reproduce. Also, looked through code and can't find the cause. Can anybody send me a request file (as I see that both of you used one) which could help me to reproduce this (to `miroslav@sqlmap.org`)? If this is a generic issue, then you could try to craft a request against a dummy site (e.g. `http://testphp.vulnweb.com`) which could provoke this. 

There was a similar issue a long time ago, but it was also fixed at that time

**Comment by stamparm** [comment]
p.s. made an update which will give you a complete traceback dump (instead of alone dummy `unable to access item...`) for your original try too. This could help too

**Comment by stamparm** [comment]
Fixed with #1880

**Comment by martindube** [comment]
Seems fixed on my side! Thank you!
